packages/react-components/package.json

```diff
@@ -6,16 +6,33 @@
     "author": "Groupe Sharegate inc.",
     "license": "Apache-2.0",
     "main": "dist/index.js",
+    "types": "dist/index.d.ts",
+    "files": [
+        "/dist",
+        "!/dist/**/docs/*",
+        "!/dist/**/tests/*"
+    ],
     "publishConfig": {
         "access": "public"
     },
     "scripts": {
         "start": "yarn watch",
         "prebuild": "yarn clean",
-        "build": "babel src -d dist --config-file ./babel.config.js --copy-files",
+        "build": "yarn run build:js",
+        "build:types": "tsc --emitDeclarationOnly",
+        "build:js": "babel src -d dist --config-file ./babel.config.js --copy-files --extensions .js,.jsx,.ts,.tsx ",
         "prewatch": "yarn clean",
         "watch": "run-p \"build --watch\"",
-        "clean": "rimraf dist"
+        "clean": "rimraf dist",
+        "type-check": "tsc --noEmit",
+        "type-check:watch": "yarn run type-check -- --watch"
```

Comment:
you don't to specify "run" with yarn, it's implicit

Reply:
what's the meaning of "type-check" and "type-check:watch" ? what's their use case? when would use use them?